src/test/java/core/basesyntax/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,413 @@
+package core.basesyntax.bookstore.controller;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import core.basesyntax.bookstore.dto.book.BookDto;
+import core.basesyntax.bookstore.dto.book.BookParamsDto;
+import core.basesyntax.bookstore.dto.book.BookWithoutCategoryDto;
+import core.basesyntax.bookstore.dto.book.CreateBookRequestDto;
+import core.basesyntax.bookstore.dto.book.UpdateBookRequestDto;
+import core.basesyntax.bookstore.dto.exception.ExceptionDto;
+import core.basesyntax.bookstore.dto.exception.ValidationExceptionDto;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MvcResult;
+
+@Sql(scripts = {
+        "classpath:database/book/remove_all_books.sql",
+        "classpath:database/category/remove_all_categories.sql",
+        "classpath:database/category/insert_2_categories.sql",
+        "classpath:database/book/insert_3_books.sql"
+},
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_CLASS)
+@Sql(scripts = {
+        "classpath:database/book/remove_all_books.sql",
+        "classpath:database/category/remove_all_categories.sql"
+},
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_CLASS)
+public class BookControllerTest extends BaseControllerTest {
+    private static final Long ID_AFTER_LAST_INSERTED_BOOK = 4L;
+    private static final String TITLE = "Shining";
+    private static final String AUTHOR = "Stephen King";
+    private static final BigDecimal PRICE = BigDecimal.valueOf(100);
+    private static final Long CATEGORY_ID = 1L;
+    private static final String ISBN = "12345678";
+    private static final String DESCRIPTION = "spooky book";
+    private static final String COVER_IMAGE = "www.cool-images.com/shining";
+
+    @WithMockUser(username = "user", roles = "USER")
+    @SneakyThrows
+    @Test
+    void findAll_getAllBooksFromDb_200() {
```

Comment:
it makes more sense in controller tests to use expected status code instead of just ok/notOk since it's more informative and practically same in terms of method name length